WordPressAuthenticator/OAuth/ProofKeyForCodeExchange.swift

```diff
@@ -25,21 +25,66 @@ struct ProofKeyForCodeExchange {
         }
     }
 
-    let codeVerifier: String
+    let codeVerifier: CodeVerifier
     let method: Method
 
-    init(codeVerifier: String, method: Method) {
+    init(codeVerifier: CodeVerifier = CodeVerifier(), method: Method = .s256) {
         self.codeVerifier = codeVerifier
         self.method = method
     }
 
     var codeCallenge: String {
         switch method {
         case .s256:
-            // TODO: code_challenge = BASE64URL-ENCODE(SHA256(ASCII(code_verifier)))
-            fatalError()
+            // The spec defines code_challenge for the s256 mode as:
+            //
+            // code_challenge = BASE64URL-ENCODE(SHA256(ASCII(code_verifier)))
+            //
+            // We don't need the ASCII conversion, because we build `CodeVerifier` from URL safe
+            // characters.
+            let rawData = codeVerifier.rawValue.data(using: .utf8)!
```

Comment:
Is it safe to force unwrap?

Reply:
You shouldn't need to!

```suggestion
            let rawData = Data(codeVerifier.rawValue.utf8)
```

Reply:
😍 

Thanks!

See 1adc8564